### PR TITLE
Update Control GUI for Arbitrary Thrusters and Rounding

### PR DESCRIPTION
**Pull request by ryan-summers**
This pull request updates the control GUI to function with an arbitrary number of thrusters in a thruster packet. Additionally, it makes it so that Qt rounds the floats to the proper number of decimal points. 
This has been dev tested using the messenger for sending packets.

### CODE REVIEW

File: gui/src/control/sensors.cpp [hunk 1]
```diff
@@ -41,14 +41,16 @@ float Sensors::average_deque(deque <float> &que)
...
 {
-    time_t t_start, t_finish;
     while (!stopped)
 	{
-        t_start = clock();
 		//Update the sensors
 
 		//Receive messages from the communicator
-		communicator->receive_messages();
+        vector<string> msgs = communicator->receive_messages();
```
Comment (ryan-summers) [inline gui/src/control/sensors.cpp]: Needs to be removed. It was originally for debug purposes.

File: gui/include/control/Sensors.hpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ class Sensors : public QThread
 signals:
 	void processedSensors(int pitch, int roll, int yaw, float depth, float battery, bool start_switch);
 	void processedRawSensors(int accel_x, int accel_y, int accel_z, float depth, int mag_x, int mag_y, int mag_z, int temp, int gyro_x, int gyro_y, int gyro_z);
-    void processedThrusterCommands(double xa, double xb, double ya, double yb, double za, double zb);
+    void processedThrusterCommands(double *commands, int len);
```
Comment (irwineffect) [inline gui/include/control/Sensors.hpp]: Any reason not to use a vector here?

Comment (ryan-summers) [inline gui/include/control/Sensors.hpp]: Vectors and the STL do not agree with Qt's signals and slots system. It is very bad at registering typed signals to typed slots. Also, this is actually two threads attempting to communicate. A static array is used to ensure that memory remains in scope as a pointer is passed to another thread.

Comment (irwineffect) [inline gui/include/control/Sensors.hpp]: Sounds good

**Comment by irwineffect** [comment]
This looks good to me. One more approval and I'll pull it into dev.

**Pull request merged**